finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/Message.scala

```diff
@@ -60,7 +61,8 @@ object Headers {
     def toSeq = synchronized(current)
     override def toString = s"""Headers(${toSeq.mkString(", ")})"""
     def contains(key: String) = toSeq.exists { case (k, _) => key == k }
-    def get(key: String) = toSeq.collect { case (k, v) if key == k => v }
+    def get(key: String) = toSeq.collectFirst { case (k, v) if key == k => v }
```

Comment:
Ah, ok -- thanks for the clarification. Will revert to the collect methods, but we can revisit later if this remains a hotspot.